controllers/storagecluster/reconcile.go

```diff
@@ -143,6 +147,17 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 	r.Log = r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	r.ctx = ctrllog.IntoContext(ctx, r.Log)
 
+	crd := &metav1.PartialObjectMetadata{}
+	crd.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+	crd.Name = VirtualMachineCrdName
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(crd), crd)
+	if client.IgnoreNotFound(err) != nil {
+		r.Log.Error(err, "Failed to get CRD", "CRD", VirtualMachineCrdName)
+		return reconcile.Result{}, err
+	}
+	crdExist := err == nil
+	statusutil.AssertEqual(r.AvailableCrds[VirtualMachineCrdName], crdExist, statusutil.RestartExitCode)
```

Comment:
Similar comment to the one I left on the ocs initialization controller on a similar code pattern